src/EnergyPlus/ScheduleManager.cc

```diff
@@ -4491,6 +4491,61 @@ namespace ScheduleManager {
         return MaximumValue;
     }
 
+    std::pair<Real64, Real64> getScheduleMinMaxByDayType(EnergyPlusData &state, int const ScheduleIndex, DayTypeGroup const days)
+    {
+        // J. Glazer - March 2024
+        // finds the minimum and maximum for a specific set of day types for a given schedule
+        Real64 MinValue = Constant::BigNumber;
+        Real64 MaxValue = -Constant::BigNumber;
+        //  Sun    Mon   Tues  Wed   Thur  Fri   Sat    Hol    Summer Winter Cust1  Cust2
+        constexpr std::array<bool, maxDayTypes> dayTypeFilterWkDy = {false, true, true, true, true, true, false, false, false, false, false, false};
+        constexpr std::array<bool, maxDayTypes> dayTypeFilterWeHo = {true, false, false, false, false, false, true, true, false, false, false, false};
+        constexpr std::array<bool, maxDayTypes> dayTypeFilterDsDy = {
+            false, false, false, false, false, false, false, false, true, true, false, false};
+        constexpr std::array<bool, maxDayTypes> dayTypeFilterNone = {
+            false, false, false, false, false, false, false, false, false, false, false, false};
+        if (ScheduleIndex == -1) {
+            MinValue = 1.0;
+            MaxValue = 1.0;
+        } else if (ScheduleIndex == 0) {
+            MinValue = 0.0;
+            MaxValue = 0.0;
+        } else if (ScheduleIndex < 1 || ScheduleIndex > state.dataScheduleMgr->NumSchedules) {
+            ShowFatalError(state, "getScheduleMinMaxByDayType called with ScheduleIndex out of range");
+        } else if (ScheduleIndex > 0) {
+            std::array<bool, maxDayTypes> dayTypeFilter;
+            std::fill(dayTypeFilter.begin(), dayTypeFilter.end(), false);
```

Comment:
Is this initialization (4517) really necessary if you are going to fill this array anyway with one of the below? Also, ScheduleIndex < 1 (line 4513) is after testing for ScheduleIndex == -1 (4507) and == 0 (4510). Is it even possible for a schedule index to be less than -1? if not, then ScheduleIndex < 1 at line 4515 is not needed (not sure that fatal is needed either but it's safe). Ideally, if the fatal were not there then `if (ScheduleIndex > 0)` could be first in the IF ladder since it is most common. This is probably way too picky for something that has limited use for the new table columns.

Reply:
@rraustad thanks for the careful review. I think I addressed your suggestions with the latest comit.